express_relay/programs/express_relay/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+
+        express_relay_metadata_data.admin = *ctx.accounts.admin.key;
+        express_relay_metadata_data.relayer_signer = *ctx.accounts.relayer_signer.key;
+        express_relay_metadata_data.fee_receiver_relayer = *ctx.accounts.fee_receiver_relayer.key;
+        express_relay_metadata_data.split_protocol_default = data.split_protocol_default;
+        express_relay_metadata_data.split_relayer = data.split_relayer;
+
+        Ok(())
+    }
+
+    pub fn set_admin(ctx: Context<SetAdmin>) -> Result<()> {
+        let express_relay_metadata_data = &mut ctx.accounts.express_relay_metadata;
+
+        express_relay_metadata_data.admin = *ctx.accounts.admin_new.key;
+
+        Ok(())
+    }
+
+    pub fn set_relayer(ctx: Context<SetRelayer>) -> Result<()> {
+        let express_relay_metadata_data = &mut ctx.accounts.express_relay_metadata;
+
+        express_relay_metadata_data.relayer_signer = *ctx.accounts.relayer_signer.key;
+        express_relay_metadata_data.fee_receiver_relayer = *ctx.accounts.fee_receiver_relayer.key;
+
+        Ok(())
+    }
+
+    pub fn set_splits(ctx: Context<SetSplits>, data: SetSplitsArgs) -> Result<()> {
+        validate_fee_split(data.split_protocol_default)?;
+        validate_fee_split(data.split_relayer)?;
+
+        let express_relay_metadata_data = &mut ctx.accounts.express_relay_metadata;
+
+        express_relay_metadata_data.split_protocol_default = data.split_protocol_default;
+        express_relay_metadata_data.split_relayer = data.split_relayer;
+
+        Ok(())
+    }
+
+    pub fn set_protocol_split(ctx: Context<SetProtocolSplit>, data: SetProtocolSplitArgs) -> Result<()> {
+        validate_fee_split(data.split_protocol)?;
+
+        ctx.accounts.protocol_config.protocol = *ctx.accounts.protocol.key;
+        ctx.accounts.protocol_config.split = data.split_protocol;
+
+        Ok(())
+    }
+
+    pub fn permission(ctx: Context<Permission>, data: PermissionArgs) -> Result<()> {
+        if data.deadline < Clock::get()?.unix_timestamp {
+            return err!(ErrorCode::DeadlinePassed);
+        }
+
+        // check that not cpi
+        if get_stack_height() > TRANSACTION_LEVEL_STACK_HEIGHT {
+            return err!(ErrorCode::InvalidCPIPermission);
+        }
+
+        // handle bid payment
+        let bid_amount = data.bid_amount;
+        let searcher = &ctx.accounts.searcher;
+        let rent_searcher = Rent::get()?.minimum_balance(searcher.to_account_info().data_len());
+        if bid_amount + rent_searcher > searcher.lamports() {
+            return err!(ErrorCode::InsufficientSearcherFunds);
+        }
+
+        let express_relay_metadata = &ctx.accounts.express_relay_metadata;
+        let split_relayer = express_relay_metadata.split_relayer;
+        let split_protocol_default = express_relay_metadata.split_protocol_default;
+
+        let split_protocol: u64;
+        let protocol_config = &ctx.accounts.protocol_config;
```

Comment:
You can use the anchor feature `Option<Account<'info,ConfigProtocol>` for this case

Reply:
nice will take a look

Reply:
i think this actually doesn't work here unfortunately. even when we aren't using this account bc the protocol doesn't have a `protocol_config`, we still need to check it to make sure that a user hasn't just excluded it from the instruction.